src/main.rb

```diff
@@ -22,29 +23,38 @@ def main(db_data)
   lists = Lists.new
   creates = Creates.new
   mockdata = MockData.new
-
-  case menu_display
-  when '1'
-    lists.list_all_books(db_data[:books])
-  when '2'
-    lists.list_all_people(db_data[:persons])
-  when '3'
-    creates.create_person(db_data[:persons])
-  when '4'
-    creates.create_book(db_data[:books])
-  when '5'
-    creates.create_rental(db_data)
-  when '6'
-    lists.list_rentals(db_data[:persons])
-  when '7'
-    puts "\nThank you for using this app!\n\n"
-    return
-  when '88'
-    mockdata.init_books_and_persons(db_data[:books], db_data[:persons])
-  else
-    puts "\nSelection not valid"
+  storage = Storage.new
+  storage.restore_all(db_data)
+  loop do
+    case menu_display
+    when '1'
+      lists.list_all_books(db_data[:books])
+    when '2'
+      lists.list_all_people(db_data[:persons])
+    when '3'
+      creates.create_person(db_data[:persons])
+    when '4'
+      creates.create_book(db_data[:books])
+    when '5'
+      creates.create_rental(db_data)
+    when '6'
+      lists.list_rentals(db_data[:persons])
+    when '7'
+      storage.save_all(db_data)
+      puts "\nSaving the database!\n"
+      puts "\nThank you for using this app!\n\n"
+      return
+    when '88'
+      mockdata.init_books_and_persons(db_data[:books], db_data[:persons])
+    when '99'
+      storage.save_all(db_data)
+    when '100'
+      storage.restore_all(db_data)
+    else
+      puts "\nSelection not valid"
+    end
   end
-  main(db_data)
+  # main(db_data)
```

Comment:
- [ ] Kindly remove all commented out, un used. unwanted pieces of code from your code base. It makes it look messy and cluttered.

Reply:
@Nemwel-Boniface Ok. I did it !